### PR TITLE
[CONTRIB] - Add athena credential support

### PR DESCRIPTION
**Pull request by tmilitino**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by tmilitino** [comment]
if the core team is also working on something similar let me know

File: great_expectations/cli/datasource.py [hunk 1]
```diff
@@ -746,6 +747,46 @@ def credentials_snippet(self) -> str:
 table_name = "{self.table_name}"'''
 
 
+class AthenaCredentialYamlHelper(SQLCredentialYamlHelper):
+    def __init__(self, datasource_name: Optional[str]) -> None:
+        # We are insisting on psycopg2 driver when adding a Redshift datasource
+        # through the CLI to avoid over-complication of this flow. If user wants
+        # to use another driver, they must use a sqlalchemy connection string.
```
Comment (anthonyburdi) [inline great_expectations/cli/datasource.py]:
```suggestion
```

**Review by anthonyburdi — APPROVED** [review]
Thank you for this contribution!

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Apologies - approved too fast. There is a failing test to address first, I also see this is marked as `draft`, and also please see my suggestion to remove the extraneous comment. But otherwise it does look good! This is the failing test: `test_cli_datasource_new_connection_string` (here is the CI log: https://dev.azure.com/great-expectations/great_expectations/_build/results?buildId=193049&view=logs&j=a5c71d55-c32d-50bf-9200-1f4fdc7a9682&t=68b39377-4f12-5f97-f5c6-79d1a0b1943b&l=820)

**Comment by tmilitino** [comment]
@anthonyburdi my apologies for the inconvenience. I sent this pr to check if the core team is working on a similar issue. I'll consider your comments and add some more unit tests. Thank you very much.

File: great_expectations/cli/datasource.py (hunk 1 above)
Comment (tmilitino) [reply great_expectations/cli/datasource.py]: done

**Pull request closed without merging**